banyand/liaison/http/server.go

```diff
@@ -134,23 +146,148 @@ func (p *server) GetPort() *uint32 {
 
 func (p *server) PreRun(_ context.Context) error {
 	p.l = logger.GetLogger(p.Name())
+	p.l.Info().Str("level", p.l.GetLevel().String()).Msg("Logger initialized")
+
+	// Log flag values after parsing
+	p.l.Debug().Bool("tls", p.tls).Str("certFile", p.certFile).Str("keyFile", p.keyFile).Msg("Flag values after parsing")
+
+	// Initialize TLSReloader if TLS is enabled
+	p.l.Debug().Bool("tls", p.tls).Msg("HTTP TLS flag is set")
+	if p.tls {
+		p.l.Debug().Str("certFile", p.certFile).Str("keyFile", p.keyFile).Msg("Initializing TLSReloader for HTTP")
+		var err error
+		p.tlsReloader, err = pkgtls.NewReloader(p.certFile, p.keyFile, p.l)
+		if err != nil {
+			p.l.Error().Err(err).Msg("Failed to initialize TLSReloader for HTTP")
+			return err
+		}
+	} else {
+		p.l.Warn().Msg("HTTP TLS is disabled, skipping TLSReloader initialization")
+	}
+
+	// Initialize gRPC client with cert file
+	if p.grpcCert != "" {
+		p.l.Debug().Str("grpcCert", p.grpcCert).Msg("Initializing TLS credentials for gRPC connection")
+
+		// Load the client cert directly - for client certs we don't need a reloader
+		// as we only need the public cert to verify the server's identity
+		cert, err := os.ReadFile(p.grpcCert)
+		if err != nil {
+			p.l.Error().Err(err).Msg("Failed to read gRPC cert file")
+			return err
+		}
+
+		certPool := x509.NewCertPool()
+		if !certPool.AppendCertsFromPEM(cert) {
+			p.l.Error().Msg("Failed to append gRPC cert to pool")
+			return errors.New("failed to append gRPC cert to pool")
+		}
+
+		// Extract hostname from grpcAddr
+		host, _, err := net.SplitHostPort(p.grpcAddr)
+		if err != nil {
+			p.l.Error().Err(err).Msg("Failed to split gRPC address")
+			return err
+		}
+		if host == "" || host == "0.0.0.0" || host == "[::]" {
+			host = "localhost"
+		}
+
+		p.creds = credentials.NewTLS(&tls.Config{
```

Comment:
The new credentials are not applied to the client because they are created statically.

Your test passed, but it cannot verify that the credentials are updated. The gRPC connection between the HTTP server and the gRPC server is long-lived. Even if the server's credentials are updated, the connection does not close and recreate itself, which is why your test does not fail.